src/st_stuff.c

```diff
@@ -271,1207 +121,1487 @@ int armor_green;   // armor amount above is blue, below is green
+    {
+        elem->xoffset -= (totalwidth >> 1);
+    }
+    else if (elem->alignment & sbe_h_right)
     {
-      if (!(plyr->keyblinktics & (2*KEYBLINKMASK - 1)))
-        S_StartSoundPitch(NULL, sfx_itemup, PITCH_NONE);
+        elem->xoffset -= totalwidth;
+    }
 
-      plyr->keyblinktics--;
+    elem->hudfont = font;
+    elem->totalwidth = totalwidth;
+}
 
-      for (i = 0; i < 3; i++)
-      {
-        switch (ST_BlinkKey(plyr, i))
+static void UpdateAnimation(sbarelem_t *elem)
+{
+    if (elem->duration_left == 0)
+    {
+        ++elem->frame_index;
+        if (elem->frame_index == array_size(elem->frames))
         {
-          case KEYBLINK_NONE:
-            continue;
+            elem->frame_index = 0;
+        }
+        elem->duration_left = elem->frames[elem->frame_index].duration;
+    }
 
-          case KEYBLINK_CARD:
-            keyboxes[i] = i;
-            break;
+    --elem->duration_left;
+}
+
+static void UpdateBoomColors(sbarelem_t *elem, player_t *player)
+{
+    if (!sts_colored_numbers)
+    {
+        elem->crboom = CR_NONE;
+        return;
+    }
 
-          case KEYBLINK_SKULL:
-            keyboxes[i] = i + 3;
+    boolean invul = (player->powers[pw_invulnerability]
+                     || player->cheats & CF_GODMODE);
+
+    crange_idx_e cr;
+
+    switch (elem->numtype)
+    {
+        case sbn_health:
+            {
+                int health = player->health;
+                if (invul)
+                    cr = CR_GRAY;
+                else if (health < health_red)
+                    cr = CR_RED;
+                else if (health < health_yellow)
+                    cr = CR_GOLD;
+                else if (health <= health_green)
+                    cr = CR_GREEN;
+                else
+                    cr = CR_BLUE2;
+            }
+            break;
+        case sbn_armor:
+            if (hud_armor_type)
+            {
+                if (invul)
+                    cr = CR_GRAY;
+                else if (!player->armortype)
+                    cr = CR_RED;
+                else if (player->armortype == 1)
+                    cr = CR_GREEN;
+                else
+                    cr = CR_BLUE2;
+            }
+            else
+            {
+                int armor = player->armorpoints;
+                if (invul)
+                    cr = CR_GRAY;
+                else if (armor < armor_red)
+                    cr = CR_RED;
+                else if (armor < armor_yellow)
+                    cr = CR_GOLD;
+                else if (armor <= armor_green)
+                    cr = CR_GREEN;
+                else
+                    cr = CR_BLUE2;
+            }
             break;
+        case sbn_ammoselected:
+            {
+                ammotype_t type = weaponinfo[player->readyweapon].ammo;
+                if (type == am_noammo)
+                {
+                    return;
+                }
+
+                int maxammo = player->maxammo[type];
```

Comment:
Please treat `player->maxammo[type] == 0` the same as `type == am_noammo`.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1916/commits/692c436a252ca74ad669ffb0f5fd27e13e222711